hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1290,14 +1323,16 @@ func machineTemplateBuilders(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.
 		mutateTemplate = func(object client.Object) error {
 			o, _ := object.(*capiaws.AWSMachineTemplate)
 			o.Spec = *machineTemplateSpec.(*capiaws.AWSMachineTemplateSpec)
+			o.SetAnnotations(map[string]string{nodePoolAnnotation: client.ObjectKeyFromObject(nodePool).String()})
```

Comment:
we need to set only the particular nodePoolAnnotation annotation here (not the whole slice) since there might be other (capi) controllers setting annotations here that we don't want to stomp.

Reply:
Good point, fixed